### DISCUSSION TITLE
usePathname / getPathname in React Server Components (RSC)

### DISCUSSION THREAD

**Discussion by alexdln** [vercel/next.js#65282] [App Router]
Just now, @delbaoliveira gave a [detailed answer](https://github.com/vercel/next.js/issues/43704#issuecomment-2090798307) on why it's not possible to get the path in Layout and RSC (_many thanks for that, team position responses are very important!_).

But I still have a question. In your responses over and over again, the reason is explained as "**layout does not re-render**". Now `PPR segments` will not be re-rendered either.

But this implies only one consequence - pathname should not be accessible within layout and PPR segments. So why not go the way of blocking the path retrieval API **only** in these abstractions?

The current situation suggests that a static site in next.js is incomplete.

I'll explain with a specific example why this is needed:
1. There is a static site. It has a fairly large and complex block, let's say pricing. It is located at dozens of levels of nesting and let's say at the eighth level we need a condition from the region - either we show tariffs `ABCD`, or tariffs `ABCE`. At the same time, they have a large volume of data and walking to the server for information. Therefore, I want to have them specifically as a server component.
1. Another example is, of course, translations (_more information in [other discussion](https://github.com/vercel/next.js/discussions/58862)_). Simply there is no way in static sites to understand deep down in server components what the current locale is. _The solution with cache for this is interesting, but doubtful._

_With respect, the author of mentioned in the closed task [nimpl-getters](https://github.com/vordgi/nimpl-getters), containing this functionality!_

**Comment by crawlchange** [comment]
She did give a detailed answer regarding the layout.

What about templates, then?
